process/block/preprocess/transactions.go

```diff
@@ -471,9 +473,175 @@ func (txs *transactions) CreateAndProcessMiniBlocks(
 		}
 	}
 
+	compressedMiniBlocks := txs.compressMiniBlocks(miniBlocks)
+	if len(compressedMiniBlocks) > 0 {
+		return compressedMiniBlocks, nil
+	}
+
 	return miniBlocks, nil
 }
 
+func (txs *transactions) compressMiniBlocks(miniBlocks block.MiniBlockSlice) block.MiniBlockSlice {
+	if len(miniBlocks) == 0 {
+		return miniBlocks
+	}
+
+	compressedMbs := make(block.MiniBlockSlice, 0)
+	compressedMbs = append(compressedMbs, miniBlocks[0])
+
+	for i := 1; i < len(miniBlocks); i++ {
+		compressedMbs = txs.merge(compressedMbs, miniBlocks[i])
+	}
+
+	log.Info(fmt.Sprintf("compressed from %d miniblocks to %d miniblocks", len(miniBlocks), len(compressedMbs)))
+
+	return compressedMbs
+}
+
+func (txs *transactions) merge(mergedMiniBlocks block.MiniBlockSlice, miniBlock *block.MiniBlock) block.MiniBlockSlice {
+	// todo: check gas limit
```

Comment:
Do not forget to implement the check for gas limit

Reply:
this is the most important part. add it before merging.